### DISCUSSION TITLE
How can i get a FakePlayer after the callback?

### DISCUSSION THREAD

**Discussion by Jochengehtab** [Minestom/Minestom#1778] [Help] [answered]
Hi, i am trying to get the Fakeplayer after the callback.

**Comment by DevBlokJochem** [comment]
Hey, you can provide a UUID when initializing the player. You can create the UUID first, and save that uuid. After you created the fakeplayer, you can get that player with that uuid.

**Reply by Jochengehtab** [reply]
Yes i now, but how can i exactly become the Fakeplayer because you can not call the constructor.

**Reply by Tofaa2** [reply]
the constructor is private meaning it can be overwritten if your class extends fake player, on top of that the fake player can be fetched from the ConnectionManager so

`
FakePlayer.initPlayer(UUID.randomUuid(), "playerName", options, null);

FakePlayer player = (FakePlayer) MinecraftServer.getConnectionManager().getPlayer("playerName") // Im not too sure about the getPlayer, it might be called something else

**Comment by Jochengehtab** [comment, marked as answer]
Hello i have solve the Problem.
First you must save the uuid for example in a JSON - File

`FakePlayer.initPlayer(uuid, "Protector", fakePlayerOption, fakePlayer -> {
            fakePlayer.spawn();
            fakePlayer.setUuid(UUID.randomUUID());
            Manager.set("./ProtectorPos.json", "uuid", String.valueOf(fakePlayer.getUuid()));
            fakePlayer.teleport(pos);
        });`

For this i am using my own SaveManager so dont worry.

Then you can get the Player like this:

` JSONObject jsonObject = Manager.getHoleFile("./ProtectorPos.json");
        UUID u = UUID.fromString((String) jsonObject.get("uuid"));
        Entity fakePlayer= Entity.getEntity(u);`

So there is your fakeplayer
